server/graphql/vaccine_course/src/mutations/update_vaccine_course.rs

```diff
@@ -0,0 +1,147 @@
+use async_graphql::*;
+use graphql_core::{
+    simple_generic_errors::DatabaseError,
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    vaccine_course::update::{
+        UpdateVaccineCourse, UpdateVaccineCourseError as ServiceError, VaccineCourseItem,
+        VaccineCourseSchedule,
+    },
+};
+
+use graphql_types::types::vaccine_course::VaccineCourseNode;
+
+pub fn update_vaccine_course(
+    ctx: &Context<'_>,
+    store_id: &str,
+    input: UpdateVaccineCourseInput,
+) -> Result<UpdateVaccineCourseResponse> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::ServerAdmin,
```

Comment:
```suggestion
            resource: Resource::ServerAdmin, // https://github.com/msupply-foundation/open-msupply/issues/4057
```
@mark-prins we need to have a chat about permissions ASAP